### ISSUE TITLE
Something overrides hwdb keyboard remaps inside of X session

### ISSUE THREAD

**Issue by ia** [systemd/systemd#14315]
**systemd version the issue has been seen with**
> 
```
$ dpkg -s systemd | grep '^Version:'
Version: 237-3ubuntu10.33
```

**Used distribution**
 >
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic
# MATE derivative if it's important
```

**Hardware**
> ThinkPad X240 laptop with builtin keyboard (maybe it's important)

**Expected behaviour you didn't see**
 > custom keyboard remaps should work systemwide

**Unexpected behaviour you saw**
 > custom keyboard remaps work inside of TTY sessions (those ones which are available through Ctrl + Shift + F 1..n) but not inside of X session

**Steps to reproduce the problem**
TL;DR:
- set custom remap through hwdb configuration file
- run commands so changes would take effect:
```
$ sudo udevadm hwdb --update && sudo udevadm control --reload-rules && sudo udevadm trigger
```
- make sure that `evtest` shows proper remaps now
- make sure that `xev` shows old remaps now

----

Now, the detailed version. Sorry if it will be too detailed but I guess that it's better than no any details at all.

I just wanted to swap PgUp/PgDn with Home/End (and enabling End as End when FnLock is enabled) on the laptop. I tried to do this using ~/.Xmodmap first but it didn't work properly. Then I found out about hwdb configuration. Here is my current configuration file:
```
$ cat /etc/udev/hwdb.d/61-ThinkPad-X240-keyboard.hwdb:
# ThinkPad X240:
# - switch End and Insert keys (so that when Fn-Lock is enabled, End works without Fn).
# - switch Home/End <> PgUp/PgDn
evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX240:*
 KEYBOARD_KEY_c7=pageup
 KEYBOARD_KEY_cf=insert
 KEYBOARD_KEY_c9=home
 KEYBOARD_KEY_d1=end
 KEYBOARD_KEY_d2=pagedown
```
And it did work perfectly. Until the recent `apt dist-upgrade`. Now **it works in TTY only**. Here is the proofs - here is the logs from GUI-based `mate-terminal` inside X session:

```
$ sudo evtest /dev/input/by-path/platform-i8042-serio-0-event-kbd

===> [pressing PgUp above LeftArrow]

Event: time 1576021876.063189, -------------- SYN_REPORT ------------
^[[5~Event: time 1576021876.524882, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576021876.524882, type 1 (EV_KEY), code 102 (KEY_HOME), value 1
Event: time 1576021876.524882, -------------- SYN_REPORT ------------
Event: time 1576021876.587770, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576021876.587770, type 1 (EV_KEY), code 102 (KEY_HOME), value 0
Event: time 1576021876.587770, -------------- SYN_REPORT ------------
^[[5~Event: time 1576021876.857851, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576021876.857851, type 1 (EV_KEY), code 102 (KEY_HOME), value 1
Event: time 1576021876.857851, -------------- SYN_REPORT ------------
Event: time 1576021876.920816, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576021876.920816, type 1 (EV_KEY), code 102 (KEY_HOME), value 0
Event: time 1576021876.920816, -------------- SYN_REPORT ------------

# Yes, it's registrating KEY_HOME as it should

===> [pressing PgDn above RightArrow]

Event: time 1576021931.761434, -------------- SYN_REPORT ------------
^[[6~Event: time 1576021931.880086, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576021931.880086, type 1 (EV_KEY), code 107 (KEY_END), value 1
Event: time 1576021931.880086, -------------- SYN_REPORT ------------
Event: time 1576021931.926910, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576021931.926910, type 1 (EV_KEY), code 107 (KEY_END), value 0
Event: time 1576021931.926910, -------------- SYN_REPORT ------------
^[[6~Event: time 1576021932.037671, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576021932.037671, type 1 (EV_KEY), code 107 (KEY_END), value 1
Event: time 1576021932.037671, -------------- SYN_REPORT ------------
Event: time 1576021932.092536, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576021932.092536, type 1 (EV_KEY), code 107 (KEY_END), value 0
Event: time 1576021932.092536, -------------- SYN_REPORT ------------

# It works for PgDn as well!
```

But here is the log from `xev` when I'm pressing exactly the same pair of the physical keys (PgUp/PgDn):
```
$ xev

===> pressing PgUp:

KeyPress event, serial 56, synthetic NO, window 0x5400001,
    root 0x112, subw 0x0, time 1791814, (-320,157), root:(1041,606),
    state 0x0, keycode 81 (keysym 0xff9a, KP_Prior), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 57, synthetic NO, window 0x5400001,
    root 0x112, subw 0x0, time 1791822, (-320,157), root:(1041,606),
    state 0x0, keycode 81 (keysym 0xff9a, KP_Prior), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

# Now it's Prior while it must be KP_Home, let's test PgDn:

===> pressing PgDn:

KeyPress event, serial 41, synthetic NO, window 0x5400001,
    root 0x112, subw 0x0, time 1893432, (108,71), root:(979,534),
    state 0x0, keycode 89 (keysym 0xff9b, KP_Next), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 42, synthetic NO, window 0x5400001,
    root 0x112, subw 0x0, time 1893440, (108,71), root:(979,534),
    state 0x0, keycode 89 (keysym 0xff9b, KP_Next), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

# KP_Next instead of End :(
```

I've mentioned above that it's broken only after recent upgrade. Here is the list of updated packages (I know that packages like python or sg3-utils have nothing to do with that but I just want to provide precise logs) - I grabbed them from `/var/log/apt/history.log`:

```
Install:

linux-modules-4.15.0-73-generic:amd64 (4.15.0-73.82  ===>  automatic)
linux-headers-4.15.0-73-generic:amd64 (4.15.0-73.82  ===>  automatic)
linux-modules-extra-4.15.0-73-generic:amd64 (4.15.0-73.82  ===>  automatic)
libllvm9:amd64 (1:9-2~ubuntu18.04.1  ===>  automatic)
linux-headers-4.15.0-73:amd64 (4.15.0-73.82  ===>  automatic)
linux-image-4.15.0-73-generic:amd64 (4.15.0-73.82  ===>  automatic)
linux-oem-osp1-tools-5.0.0-1032:amd64 (5.0.0-1032.36  ===>  automatic)
linux-tools-5.0.0-1032-oem-osp1:amd64 (5.0.0-1032.36  ===>  automatic)


Upgrade:

libgraphicsmagick-q16-3:amd64 (1.3.28-2  ===>  1.3.28-2ubuntu0.1)
libdevmapper1.02.1:amd64 (2:1.02.145-4.1ubuntu3.18.04.1  ===>  2:1.02.145-4.1ubuntu3.18.04.2)
liblvm2cmd2.02:amd64 (2.02.176-4.1ubuntu3.18.04.1  ===>  2.02.176-4.1ubuntu3.18.04.2)
dmsetup:amd64 (2:1.02.145-4.1ubuntu3.18.04.1  ===>  2:1.02.145-4.1ubuntu3.18.04.2)
python2.7-dev:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
libpython3.6-minimal:amd64 (3.6.8-1~18.04.3  ===>  3.6.9-1~18.04)
intel-microcode:amd64 (3.20191115.1ubuntu0.18.04.1  ===>  3.20191115.1ubuntu0.18.04.2)
linux-headers-generic:amd64 (4.15.0.70.72  ===>  4.15.0.73.75)
libqt5test5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libdrm-nouveau2:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
python3-distutils:amd64 (3.6.8-1~18.04  ===>  3.6.9-1~18.04)
linux-libc-dev:amd64 (4.15.0-70.79  ===>  4.15.0-73.82)
linux-libc-dev:i386 (4.15.0-70.79  ===>  4.15.0-73.82)
xserver-common:amd64 (2:1.19.6-1ubuntu4.3  ===>  2:1.19.6-1ubuntu4.4)
python3-software-properties:amd64 (0.96.24.32.11  ===>  0.96.24.32.12)
qt5-gtk-platformtheme:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
xserver-xorg-core:amd64 (2:1.19.6-1ubuntu4.3  ===>  2:1.19.6-1ubuntu4.4)
libegl-mesa0:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
linux-crashdump:amd64 (4.15.0.70.72  ===>  4.15.0.73.75)
libsystemd0:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
libpulsedsp:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
grub-common:amd64 (2.02-2ubuntu8.13  ===>  2.02-2ubuntu8.14)
linux-image-generic:amd64 (4.15.0.70.72  ===>  4.15.0.73.75)
mdadm:amd64 (4.1~rc1-3~ubuntu18.04.2  ===>  4.1~rc1-3~ubuntu18.04.3)
pulseaudio:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
libglapi-mesa:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libvala-0.40-0:amd64 (0.40.8-0ubuntu1  ===>  0.40.17-0ubuntu1)
mesa-common-dev:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
python2.7-minimal:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
snapd:amd64 (2.40+18.04  ===>  2.42.1+18.04)
libsqlite3-0:amd64 (3.22.0-1ubuntu0.1  ===>  3.22.0-1ubuntu0.2)
libpython3.6-stdlib:amd64 (3.6.8-1~18.04.3  ===>  3.6.9-1~18.04)
libstrongswan:amd64 (5.6.2-1ubuntu2.4  ===>  5.6.2-1ubuntu2.5)
libdevmapper-event1.02.1:amd64 (2:1.02.145-4.1ubuntu3.18.04.1  ===>  2:1.02.145-4.1ubuntu3.18.04.2)
libqt5dbus5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libpython2.7:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
google-chrome-stable:amd64 (77.0.3865.120-1  ===>  78.0.3904.108-1)
python2.7:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
libqt5sql5-sqlite:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libpython3.6:amd64 (3.6.8-1~18.04.3  ===>  3.6.9-1~18.04)
python3.6:amd64 (3.6.8-1~18.04.3  ===>  3.6.9-1~18.04)
gdb:amd64 (8.1-0ubuntu3.1  ===>  8.1-0ubuntu3.2)
dmeventd:amd64 (2:1.02.145-4.1ubuntu3.18.04.1  ===>  2:1.02.145-4.1ubuntu3.18.04.2)
xserver-xorg-legacy:amd64 (2:1.19.6-1ubuntu4.3  ===>  2:1.19.6-1ubuntu4.4)
libxatracker2:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libqt5widgets5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
lvm2:amd64 (2.02.176-4.1ubuntu3.18.04.1  ===>  2.02.176-4.1ubuntu3.18.04.2)
valac-0.40-vapi:amd64 (0.40.8-0ubuntu1  ===>  0.40.17-0ubuntu1)
grub2-common:amd64 (2.02-2ubuntu8.13  ===>  2.02-2ubuntu8.14)
libpython2.7-dev:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
udev:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
libegl1-mesa:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
network-manager-gnome:amd64 (1.8.10-2ubuntu2  ===>  1.8.10-2ubuntu3)
software-properties-gtk:amd64 (0.96.24.32.11  ===>  0.96.24.32.12)
linux-signed-generic:amd64 (4.15.0.70.72  ===>  4.15.0.73.75)
libudev1:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
rsyslog:amd64 (8.32.0-1ubuntu4  ===>  8.32.0-1ubuntu4.1)
libqt5xml5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libgbm1:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libqt5printsupport5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
python3.6-minimal:amd64 (3.6.8-1~18.04.3  ===>  3.6.9-1~18.04)
apport:amd64 (2.20.9-0ubuntu7.9  ===>  2.20.9-0ubuntu7.10)
libdrm-amdgpu1:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
libglib2.0-dev-bin:amd64 (2.56.4-0ubuntu0.18.04.4  ===>  2.56.4-0ubuntu0.18.04.5)
libqt5gui5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
ulauncher:amd64 (5.3.0-0ubuntu1ppa1~bionic  ===>  5.4.0-0ubuntu1ppa1~bionic)
ubuntu-drivers-common:amd64 (1:0.5.2.4  ===>  1:0.5.2.5)
libnautilus-extension1a:amd64 (1:3.26.4-0~ubuntu18.04.4  ===>  1:3.26.4-0~ubuntu18.04.5)
grub-efi-amd64-bin:amd64 (2.02-2ubuntu8.13  ===>  2.02-2ubuntu8.14)
python3-distupgrade:amd64 (1:18.04.34  ===>  1:18.04.36)
libpulse0:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
makedumpfile:amd64 (1:1.6.5-1ubuntu1~18.04.2  ===>  1:1.6.5-1ubuntu1~18.04.3)
ubuntu-release-upgrader-core:amd64 (1:18.04.34  ===>  1:18.04.36)
strongswan-nm:amd64 (5.6.2-1ubuntu2.4  ===>  5.6.2-1ubuntu2.5)
python3-apport:amd64 (2.20.9-0ubuntu7.9  ===>  2.20.9-0ubuntu7.10)
qemu-user-static:amd64 (1:2.11+dfsg-1ubuntu7.20  ===>  1:2.11+dfsg-1ubuntu7.21)
liblvm2app2.2:amd64 (2.02.176-4.1ubuntu3.18.04.1  ===>  2.02.176-4.1ubuntu3.18.04.2)
libpulse-mainloop-glib0:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
systemd-sysv:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
libwayland-egl1-mesa:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
kdump-tools:amd64 (1:1.6.5-1ubuntu1~18.04.2  ===>  1:1.6.5-1ubuntu1~18.04.3)
libglib2.0-bin:amd64 (2.56.4-0ubuntu0.18.04.4  ===>  2.56.4-0ubuntu0.18.04.5)
libpulse-dev:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
gnome-shell-common:amd64 (3.28.4-0ubuntu18.04.2  ===>  3.28.4-0ubuntu18.04.3)
libpam-systemd:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
valac:amd64 (0.40.8-0ubuntu1  ===>  0.40.17-0ubuntu1)
linux-tools-oem-osp1:amd64 (5.0.0.1028.32  ===>  5.0.0.1032.36)
gdb-multiarch:amd64 (8.1-0ubuntu3.1  ===>  8.1-0ubuntu3.2)
libdrm2:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
libnma0:amd64 (1.8.10-2ubuntu2  ===>  1.8.10-2ubuntu3)
python3-lib2to3:amd64 (3.6.8-1~18.04  ===>  3.6.9-1~18.04)
libglib2.0-dev:amd64 (2.56.4-0ubuntu0.18.04.4  ===>  2.56.4-0ubuntu0.18.04.5)
systemd:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
libqt5core5a:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libgl1-mesa-dev:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libgl1-mesa-dri:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
grub-efi-amd64:amd64 (2.02-2ubuntu8.13  ===>  2.02-2ubuntu8.14)
libnss-systemd:amd64 (237-3ubuntu10.31  ===>  237-3ubuntu10.33)
libnm-gtk0:amd64 (1.8.10-2ubuntu2  ===>  1.8.10-2ubuntu3)
gdb-doc:amd64 (8.1-0ubuntu3.1  ===>  8.1-0ubuntu3.2)
libglib2.0-data:amd64 (2.56.4-0ubuntu0.18.04.4  ===>  2.56.4-0ubuntu0.18.04.5)
libqt5opengl5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libgl1-mesa-glx:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
apport-gtk:amd64 (2.20.9-0ubuntu7.9  ===>  2.20.9-0ubuntu7.10)
ubuntu-release-upgrader-gtk:amd64 (1:18.04.34  ===>  1:18.04.36)
grub-efi-amd64-signed:amd64 (1.93.14+2.02-2ubuntu8.13  ===>  1.93.15+2.02-2ubuntu8.14)
libpython2.7-minimal:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
unattended-upgrades:amd64 (1.1ubuntu1.18.04.11  ===>  1.1ubuntu1.18.04.13)
pulseaudio-module-bluetooth:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
libdrm-intel1:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
libsgutils2-2:amd64 (1.42-2ubuntu1  ===>  1.42-2ubuntu1.18.04.1)
sg3-utils:amd64 (1.42-2ubuntu1  ===>  1.42-2ubuntu1.18.04.1)
gdbserver:amd64 (8.1-0ubuntu3.1  ===>  8.1-0ubuntu3.2)
strongswan-libcharon:amd64 (5.6.2-1ubuntu2.4  ===>  5.6.2-1ubuntu2.5)
nautilus-data:amd64 (1:3.26.4-0~ubuntu18.04.4  ===>  1:3.26.4-0~ubuntu18.04.5)
libdrm-radeon1:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
mesa-vdpau-drivers:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
linux-firmware:amd64 (1.173.12  ===>  1.173.13)
linux-tools-common:amd64 (4.15.0-70.79  ===>  4.15.0-73.82)
libqt5network5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
libpython2.7-stdlib:amd64 (2.7.15-4ubuntu4~18.04.2  ===>  2.7.17-1~18.04)
python3-gdbm:amd64 (3.6.8-1~18.04  ===>  3.6.9-1~18.04)
pulseaudio-utils:amd64 (1:11.1-1ubuntu7.4  ===>  1:11.1-1ubuntu7.5)
u-boot-tools:amd64 (2018.07~rc3+dfsg1-0ubuntu3~18.04.1  ===>  2018.07~rc3+dfsg1-0ubuntu3~18.04.2)
libdrm-dev:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
linux-generic:amd64 (4.15.0.70.72  ===>  4.15.0.73.75)
python3-problem-report:amd64 (2.20.9-0ubuntu7.9  ===>  2.20.9-0ubuntu7.10)
libglib2.0-0:amd64 (2.56.4-0ubuntu0.18.04.4  ===>  2.56.4-0ubuntu0.18.04.5)
libqt5sql5:amd64 (5.9.5+dfsg-0ubuntu2.3  ===>  5.9.5+dfsg-0ubuntu2.4)
mesa-va-drivers:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libglx-mesa0:amd64 (19.0.8-0ubuntu0~18.04.3  ===>  19.2.1-1ubuntu1~18.04.1)
libdrm-common:amd64 (2.4.97-1ubuntu1~18.04.1  ===>  2.4.99-1ubuntu1~18.04.1)
software-properties-common:amd64 (0.96.24.32.11  ===>  0.96.24.32.12)
```

I even tried to downgrade some of them (such as X-related packages and systemd-related packages as well) but I got no luck with bringing back old working behavior.

DISCLAIMER: **I do realize** that it's probably **not a bug** in `udev` itself but I just hope that someone will point me out the further direction to investigate this because I'm very interested in fixing this issue due to unbearable text processing when Home/End keys are on the other side of the keyboard. And I'm ready to provide any other related information including debug logs and to test any proposed solutions. Please, help. Thanks in advance.

P.S. I'm sorry that it's against the rules to create a bug about not the latest stable upstream release but:
- I don't know where else to ask for help with testing & debugging
- I'm not sure that the latest version will help if something overrides hwdb settings from X session
It will be great if someone just will point out the further direction to investigate the issue.

**Comment by ia** [comment]
I just downloaded the latest stable Ubuntu MATE (19.10) and tested it from live mode - it works as it should. But still, maybe someone could point me where to look for the issue in 18.04 to fix it.

Here is the logs from 19.10 live session:

```
ubuntu-mate@ubuntu-mate:~$ dpkg -s systemd | grep '^Version'
Version: 242-7ubuntu3

ubuntu-mate@ubuntu-mate:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan

$ cat /etc/udev/hwdb.d/61-ThinkPad-X240-keyboard.hwdb
# ThinkPad X240:
# - switch End and Insert keys (so that when Fn-Lock is enabled, End works without Fn).
# - switch Home/End <> PgUp/PgDn
evdev:atkbd:dmi:bvn*:bvr*:bd*:svnLENOVO:pn*:pvrThinkPadX240:*
 KEYBOARD_KEY_c7=pageup
 KEYBOARD_KEY_cf=insert
 KEYBOARD_KEY_c9=home
 KEYBOARD_KEY_d1=end
 KEYBOARD_KEY_d2=pagedown

$ sudo udevadm hwdb --update
$ sudo udevadm control --reload-rules
$ sudo udevadm trigger

$ sudo systemctl stop systemd-udevd
$ sudo systemctl start systemd-udevd

$ sudo apt install evtest


### log out, log in


$ sudo evtest /dev/input/by-path/platform-i8042-serio-0-event-kbd

# ===> pressing physical PgUp/PgDn ...

Event: time 1576087375.067259, -------------- SYN_REPORT ------------
Event: time 1576087381.333948, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576087381.333948, type 1 (EV_KEY), code 102 (KEY_HOME), value 1
Event: time 1576087381.333948, -------------- SYN_REPORT ------------
^[[HEvent: time 1576087381.445897, type 4 (EV_MSC), code 4 (MSC_SCAN), value c9
Event: time 1576087381.445897, type 1 (EV_KEY), code 102 (KEY_HOME), value 0
Event: time 1576087381.445897, -------------- SYN_REPORT ------------
Event: time 1576087381.678756, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576087381.678756, type 1 (EV_KEY), code 107 (KEY_END), value 1
Event: time 1576087381.678756, -------------- SYN_REPORT ------------
^[[FEvent: time 1576087381.758437, type 4 (EV_MSC), code 4 (MSC_SCAN), value d1
Event: time 1576087381.758437, type 1 (EV_KEY), code 107 (KEY_END), value 0
Event: time 1576087381.758437, -------------- SYN_REPORT ------------


$ xev

# ===> pressing physical PgUp/PgDn ...

KeyPress event, serial 38, synthetic NO, window 0x3800001,
    root 0x15a, subw 0x0, time 887522, (110,92), root:(111,148),
    state 0x0, keycode 110 (keysym 0xff50, Home), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyRelease event, serial 38, synthetic NO, window 0x3800001,
    root 0x15a, subw 0x0, time 887617, (110,92), root:(111,148),
    state 0x0, keycode 110 (keysym 0xff50, Home), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyPress event, serial 38, synthetic NO, window 0x3800001,
    root 0x15a, subw 0x0, time 888294, (110,92), root:(111,148),
    state 0x0, keycode 115 (keysym 0xff57, End), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyRelease event, serial 38, synthetic NO, window 0x3800001,
    root 0x15a, subw 0x0, time 888333, (110,92), root:(111,148),
    state 0x0, keycode 115 (keysym 0xff57, End), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False

$
```

**Comment by ia** [comment]
Unbelievable. I have no idea what was going on but resetting `hwdb.bin` did help.

So, to whom it may concern or in case if someone will find this issue through search engines - try to ~~turn off and on again~~ **delete** `hwdb.bin` and **regenerate it again** - here is what I did:
```
$ sudo rm /etc/udev/hwdb.bin
$ sudo udevadm hwdb --update
$ sudo udevadm control --reload-rules
$ sudo udevadm trigger
$ sudo systemctl stop systemd-udevd
$ sudo systemctl start systemd-udevd
```
And **reboot** - it's mandatory because until I haven't rebooted my laptop `xev` showed old values.

P.S. I'm really sorry for disturbance.
